icu4c/source/common/charstr.cpp

```diff
@@ -141,6 +141,41 @@ CharString &CharString::append(const char *s, int32_t sLength, UErrorCode &error
     return *this;
 }
 
+CharString &CharString::appendNumber(int32_t number, UErrorCode &status) {
+    if (number < 0) {
+        this->append('-', status);
+        number *= -1;
+        if (U_FAILURE(status)) {
+            return *this;
+        }
+    }
+
+    if (number == 0) {
+        this->append('0', status);
+        return *this;
+    }
+
+    CharString numberStr;
+    while (number > 0) {
+        int32_t residue = number % 10;
+        number /= 10;
+        numberStr.append(residue + '0', status);
+        if (U_FAILURE(status)) {
+            return *this;
+        }
+    }
+
+    char *clonedNumber = numberStr.cloneData(status);
```

Comment:
You should append directly to `this` rather than creating this completely unnecessary and unsafe temporary variable.

Reply:
Done, I flipped the tail of the word.